src/core/client.cpp

```diff
@@ -35,16 +35,17 @@ namespace deepstream {
 
 Client::Client(const std::string &uri, WSHandler &ws_handler, ErrorHandler &error_handler)
     : p_connection_(new Connection(uri, ws_handler, error_handler, event, presence))
-    , event(std::bind(&Connection::send, p_connection_.get(), std::placeholders::_1))
-    , presence(std::bind(&Connection::send, p_connection_.get(), std::placeholders::_1))
+    , subscription_counter_(0)
+    , event(std::bind(&Connection::send, p_connection_.get(), std::placeholders::_1), subscription_counter_)
```

Comment:
this is similar to bind in javascript. It says create a function object that calls `Connection::send` with `this` being `p_connection_` and with the first parameter of the function object given as the first argument of `Connection::send`.